paseos/actors/actor_builder.py

```diff
@@ -211,6 +217,12 @@ def set_thermal_model(
             actor, SpacecraftActor
         ), "Thermal models are only supported for SpacecraftActors"
 
+        # Check if the actor already had a thermal model
+        if actor.has_thermal_model:
```

Comment:
As above. If you want, you can also make one statement for all the models somewhere.